### PR TITLE
Polish org.codehaus.groovy.runtime package documentation

### PR DESCRIPTION
**Pull request by marcwrobel**
Fix typos and improve documentation (javadoc, comments) in the org.codehaus.groovy.runtime package.

Note that the following words / expressions were preferred :

- all of the -> all the (when possible because it is easier to read),
- whether or not -> whether (when possible because it is easier to read),
- sub-directory -> subdirectory (more common, see https://www.merriam-webster.com/dictionary/subdirectory),
- meta class -> metaclass (incorrect, see https://en.wikipedia.org/wiki/Metaclass).

Trailing whitespaces were also removed in the process.

### CODE REVIEW

File: src/main/java/org/codehaus/groovy/runtime/metaclass/DefaultMetaClassInfo.java [hunk 1]
```diff
@@ -29,21 +29,21 @@ public class DefaultMetaClassInfo {
...
-    // if original boolean meta class
+    // if original boolean metaclass
     private static boolean origBoolean = true;
     // if origBoolean and withoutCustomHandle
     private static boolean origBooleanRes = true;
 
     /**
-     * @return  true if no meta class creation handle is set and if
-     *          the original boolean meta class is used.
+     * @return  true if no metaclass creation handle is set and if
```
[the remark this answers is not in the document]
Comment (marcwrobel) [reply src/main/java/org/codehaus/groovy/runtime/metaclass/DefaultMetaClassInfo.java]: Resolved.

File: src/main/java/org/codehaus/groovy/runtime/metaclass/DefaultMetaClassInfo.java [hunk 2]
```diff
@@ -154,21 +154,21 @@ public static void setOrigInt(boolean v) {
...
-    // if original Integer[] meta class
+    // if original Integer[] metaclass
     private static boolean origIntArray = true;
     // if origInt and withoutCustomHandle
     private static boolean origIntArrayWCH = true;
 
     /**
-     * @return  true if no meta class creation handle is set and if
-     *          the original integer array meta class is used.
+     * @return  true if no metaclass creation handle is set and if
```
[the remark this answers is not in the document]
Comment (marcwrobel) [reply src/main/java/org/codehaus/groovy/runtime/metaclass/DefaultMetaClassInfo.java]: Resolved.

File: src/main/java/org/codehaus/groovy/runtime/metaclass/DefaultMetaClassInfo.java [hunk 3]
```diff
@@ -179,21 +179,21 @@ public static void setOrigIntArray(boolean v) {
...
-    // if original long meta class
+    // if original long metaclass
     private static boolean origLong = true;
     // if origLong and withoutCustomHandle
     private static boolean origLongRes = true;
 
     /**
-     * @return  true if no meta class creation handle is set and if
-     *          the original long meta class is used.
+     * @return  true if no metaclass creation handle is set and if
```
[the remark this answers is not in the document]
Comment (marcwrobel) [reply src/main/java/org/codehaus/groovy/runtime/metaclass/DefaultMetaClassInfo.java]: Resolved.

File: src/main/java/org/codehaus/groovy/runtime/metaclass/DefaultMetaClassInfo.java [hunk 4]
```diff
@@ -229,36 +229,36 @@ public static void setOrigFloat(boolean v) {
...
-    // if original double meta class
+    // if original double metaclass
     private static boolean origDouble = true;
     // if origFloat and withoutCustomHandle
     private static boolean origDoubleRes = true;
 
     /**
-     * @return  true if no meta class creation handle is set and if
-     *          the original double meta class is used.
+     * @return  true if no metaclass creation handle is set and if
```
[the remark this answers is not in the document]
Comment (marcwrobel) [reply src/main/java/org/codehaus/groovy/runtime/metaclass/DefaultMetaClassInfo.java]: Resolved.

File: src/main/java/org/codehaus/groovy/runtime/metaclass/DefaultMetaClassInfo.java [hunk 5]
```diff
@@ -129,21 +129,21 @@ public static void setOrigShort(boolean v) {
...
-    // if original Integer meta class
+    // if original Integer metaclass
     private static boolean origInt = true;
     // if origInt and withoutCustomHandle
     private static boolean origIntRes = true;
 
     /**
-     * @return  true if no meta class creation handle is set and if
-     *          the original integer meta class is used.
+     * @return  true if no metaclass creation handle is set and if
```
[the remark this answers is not in the document]
Comment (marcwrobel) [reply src/main/java/org/codehaus/groovy/runtime/metaclass/DefaultMetaClassInfo.java]: Resolved.

File: src/main/java/org/codehaus/groovy/runtime/metaclass/DefaultMetaClassInfo.java [hunk 6]
```diff
@@ -54,21 +54,21 @@ public static void setOrigBool(boolean v) {
...
-    // if original Byte meta class
+    // if original Byte metaclass
     private static boolean origByte = true;
     // if origByte and withoutCustomHandle
     private static boolean origByteRes = true;
 
     /**
-     * @return  true if no meta class creation handle is set and if
-     *          the original byte meta class is used.
+     * @return  true if no metaclass creation handle is set and if
```
[the remark this answers is not in the document]
Comment (marcwrobel) [reply src/main/java/org/codehaus/groovy/runtime/metaclass/DefaultMetaClassInfo.java]: Resolved.

File: src/main/java/org/codehaus/groovy/runtime/metaclass/DefaultMetaClassInfo.java [hunk 7]
```diff
@@ -79,21 +79,21 @@ public static void setOrigByte(boolean v) {
...
-    // if original char meta class
+    // if original char metaclass
     private static boolean origChar = true;
     // if origChar and withoutCustomHandle
     private static boolean origCharRes = true;
 
     /**
-     * @return  true if no meta class creation handle is set and if
-     *          the original char meta class is used.
+     * @return  true if no metaclass creation handle is set and if
```
[the remark this answers is not in the document]
Comment (marcwrobel) [reply src/main/java/org/codehaus/groovy/runtime/metaclass/DefaultMetaClassInfo.java]: Resolved.

File: src/main/java/org/codehaus/groovy/runtime/metaclass/DefaultMetaClassInfo.java [hunk 8]
```diff
@@ -204,21 +204,21 @@ public static void setOrigLong(boolean v) {
...
-    // if original float meta class
+    // if original float metaclass
     private static boolean origFloat = true;
     // if origFloat and withoutCustomHandle
     private static boolean origFloatRes = true;
 
     /**
-     * @return  true if no meta class creation handle is set and if
-     *          the original float meta class is used.
+     * @return  true if no metaclass creation handle is set and if
```
[the remark this answers is not in the document]
Comment (marcwrobel) [reply src/main/java/org/codehaus/groovy/runtime/metaclass/DefaultMetaClassInfo.java]: Resolved.

File: src/main/java/org/codehaus/groovy/runtime/metaclass/ConcurrentReaderHashMap.java [hunk 9]
```diff
@@ -663,7 +663,7 @@ public Object remove(Object key) {
             count--;
 
             Entry head = e.next;
-            for (Entry p = first; p != e; p = p.next) 
+            for (Entry p = first; p != e; p = p.next)
```
[the remark this answers is not in the document]
Comment (marcwrobel) [reply src/main/java/org/codehaus/groovy/runtime/metaclass/ConcurrentReaderHashMap.java]: Change reverted.

**Comment by paulk-asert** [comment]
Merged, thanks!

**Comment by marcwrobel** [comment]
That was the last one, thanks for the review @paulk-asert. Hope that was not too painful ;)

**Pull request merged**